src/components/Divider.tsx

```diff
@@ -1,50 +1,81 @@
-  const dividerSetWidth = React.createRef<HTMLInputElement>()
+  const dividerRef = React.createRef<HTMLInputElement>()
   const dispatch = useDispatch()
+  const [width, setWidth] = useState(DIVIDER_MIN_WIDTH)
 
   /** Sets the cursor to the divider. */
   const setCursorToDivider = (e: React.MouseEvent | React.TouchEvent) => {
     e.stopPropagation()
     dispatch(setCursor({ path }))
   }
 
-  /** Get the max width of nearby for divider list child elements, add 30 px and set this width for divider. */
+  /** Get the max width of nearby elements, add DIVIDER_PLUS_PX and set this width for divider. */
   const setStyle = () => {
-    if (dividerSetWidth.current) {
-      const parentUl = dividerSetWidth.current.closest('ul')
-      const children = parentUl ? (Array.from(parentUl.childNodes) as HTMLElement[]) : []
-      const widths = children.map((child: HTMLElement) => {
-        if (child.classList.contains('child-divider')) return DIVIDER_PLUS_PX
-        const subs = child.getElementsByClassName('subthought') as HTMLCollectionOf<HTMLElement>
-        return subs.length ? subs[0].offsetWidth + DIVIDER_PLUS_PX : DIVIDER_PLUS_PX
-      })
-      const maxWidth = Math.max(...widths)
-      dividerSetWidth.current.style.width = `${maxWidth > DIVIDER_MIN_WIDTH ? maxWidth : DIVIDER_MIN_WIDTH}px`
+    if (dividerRef.current) {
+      const state = store.getState()
+      const parentPath = rootedParentOf(state, path)
+
+      const treeNode = dividerRef.current.closest('div.tree-node') as HTMLElement
+      if (!treeNode) throw new Error('Divider tree node not found')
+
+      const thoughtPath = treeNode.dataset.path
+      if (!thoughtPath || thoughtPath === '') throw new Error('Divider path not found on tree node')
+
+      let elements = document.querySelectorAll(`.tree-node[data-path^="${hashPath(parentPath)}"]:not(.thought-divider)`)
+
+      if (elements.length === 1) {
+        /** If this divider is an only child in a table, find the widest col2. */
+        const grandparentPath = rootedParentOf(state, parentPath)
+        if (!grandparentPath) throw new Error('Divider grandparent not found')
+
+        elements = document.querySelectorAll(
+          `.tree-node.table-col2[data-path^="${hashPath(grandparentPath)}"]:not(.thought-divider)`,
+        )
+      } else if (!elements.length) {
+        /** This is a top-level divider, use its siblings instead. */
+        elements = document.querySelectorAll('.tree-node.top-level:not(.thought-divider)')
+      }
+
+      /** Find and set the max width of our selected elements. */
+      const maxWidth = Math.max(
+        ...Array.from(elements).map(
+          element => (element.querySelector('.thought') as HTMLElement).offsetWidth + DIVIDER_PLUS_PX,
+        ),
+      )
+
+      setWidth(maxWidth)
     }
   }
 
-  useEffect(setStyle)
+  useEffect(setStyle, []) // eslint-disable-line react-hooks/exhaustive-deps
+  editingValueStore.useEffect(setStyle)
```

Comment:
I did this because we need `setStyle` to be called on the initial render before `editingValueStore` takes over, but providing the expected dependencies causes a loop and it gets called repeatedly. I've created an experimental branch to show this in action:

https://github.com/mathiscode/em/tree/experiment/543-divider-width-exhaustive-deps

I'm definitely open to any suggestions for a more optimal solution, but this does seem to solve the immediate problem.

Reply:
It looks like `dividerRef` is not a stable reference, even if `dividerRef.current` is. Learned something new! :)

So you should be able to fix it with `[dividerRef.current, path]`.